src/AzureExtension/Providers/RepositoryProvider.cs

```diff
@@ -347,96 +359,135 @@ public void Dispose()
-                        organizationsAndProjects.Add(organization, projects);
-                    }
-                });
+        Dictionary<Organization, List<TeamProjectReference>> organizationsAndProjects = new Dictionary<Organization, List<TeamProjectReference>>();
 
-                var reposToReturn = new List<IRepository>();
+        // Establish a connection between organizations and their projects.
+        Parallel.ForEach(organizations, organization =>
+        {
+            var projects = _azureHierarchy.GetProjectsAsync(organization).Result;
 
-                // organizationAndProjects include all relevant information to get repos.
-                Parallel.ForEach(organizationsAndProjects, organizationAndProject =>
-                {
-                    try
-                    {
-                        // Making a connection can throw.
-                        var connection = AzureClientProvider.GetConnectionForLoggedInDeveloper(organizationAndProject.Key.AccountUri, azureDeveloperId);
-                        var criteria = new GitPullRequestSearchCriteria();
-                        criteria.CreatorId = connection.AuthorizedIdentity.Id;
+            if (projects.Count != 0)
+            {
+                organizationsAndProjects.Add(organization, projects);
+            }
+        });
 
-                        Parallel.ForEach(organizationAndProject.Value, project =>
-                        {
-                            reposToReturn.AddRange(GetRepos(connection, project, criteria, shouldIgnorePRDateOld));
-                        });
-                    }
-                    catch (Exception e)
-                    {
-                        Providers.Log.Logger()?.ReportError("DevHomeRepository", e);
-                    }
+        var reposToReturn = new List<IRepository>();
+
+        // organizationAndProjects include all relevant information to get repos.
+        Parallel.ForEach(organizationsAndProjects, organizationAndProject =>
+        {
+            try
+            {
+                // Making a connection can throw.
+                var connection = AzureClientProvider.GetConnectionForLoggedInDeveloper(organizationAndProject.Key.AccountUri, azureDeveloperId);
+                var criteria = new GitPullRequestSearchCriteria();
+                criteria.CreatorId = connection.AuthorizedIdentity.Id;
+
+                Parallel.ForEach(organizationAndProject.Value, project =>
+                {
+                    reposToReturn.AddRange(GetRepos(connection, project, criteria, shouldIgnorePRDate));
                 });
+            }
+            catch (Exception e)
+            {
+                Providers.Log.Logger()?.ReportError("DevHomeRepository", e);
+            }
+        });
 
-                return new RepositoriesResult(reposToReturn);
+        // Figure out what server the organizations are from. if the user did not supply one.
+        if (string.IsNullOrEmpty(_serverToSearch))

```

Comment:
There is a check against _serverToSearch at the beginning of this method. Now we initialize it from Empty to something else. This seems backwards. If this is by design then, a better comment is needed to explain how _serverToSearch is supposed to be used.

Reply:
Added more comments.  This code is used to find out what server all the organizations belong to (if the user didn't search for a specific server).